mobile-browser-based-version/src/helpers/communication_script/helpers.js

```diff
@@ -44,13 +44,19 @@ export function assignWeightsToModel(serializedWeights, model) {
     });
 }
 
-export function averageWeightsIntoModel(serializedWeights, model) {
+export function averageWeightsIntoModel(peersSerializedWeights, model) {
     model.weights.forEach((weight, idx) => {
-        const serializedWeight = serializedWeights[idx]["$variable"];
-
-        const tensor = deserializeTensor(serializedWeight.val);
-        weight.val.assign(tensor.add(weight.val).div(2)); //average
-        tensor.dispose();
+        let tensorSum = weight.val
+        peersSerializedWeights.forEach(
+            (serializedWeights, peer) =>{
+                const serializedWeight = serializedWeights[idx]["$variable"];
+                const tensor = deserializeTensor(serializedWeight.val);
+                tensorSum = tensor.add(tensorSum)
+                tensor.dispose()
+            }
+        )
+        weight.val.assign(tensorSum.div(peersSerializedWeights.length)); //average
```

Comment:
I think it should be changed to 
```
weight.val.assign(tensorSum.div(peersSerializedWeights.length+1)); //average
```
because we also include the local weight in the sum

Reply:
Oh yes definitely! Thanks for pointing this out!